src/reference/informative/informC04_component_reference.rst

```diff
@@ -8,4 +8,105 @@
 
   .. container:: infospec
 
-    **TODO**
\ No newline at end of file
+    .. container:: heading3
+
+      Understanding component references
+
+    As with references to units and variables, the term "component reference" refers to both sides of the naming and calling transaction.
+    Its use includes both the item which *possesses* the reference (i.e.: a :code:`component` or :code:`import component` with a :code:`name` attribute) as well as the item which is *calling* the reference (i.e.: an item requesting the named component via its own :code:`component_ref` attribute).
```

Comment:
I don't get this. Surely if X has name "x", and I refer to "x", that referring is not part of X? If I write a book about your book, the reference in my book isn't a property of your book all of a sudden?

Reply:
When I write `<component name="x">` there's no component reference there, right? Just a component and a name?

Reply:
So I think we can agree that I have a lot of trouble understanding the "reference" sections in general!  My (latest ...) understanding is that both the name-of-the-thing and the thing-calling-the-name were "references"?  
So for `<component name="myName">` the name attribute value was the reference, and 
for `<connection component_1="myName" ...>` the value of the component_1 attribute was the reference.  This is what I was trying to say with the both sides of a naming/calling bit (though if it's not what's happening, we can ditch that part!).  I've found all of the sections about references really hard to follow ... (which could be just me missing something), but also could show that we need to write them in a much clearer way?  I am happy to be the lowest-understanding-denominator guinea-pig ... but I can't make too many suggestions without actually understanding it first!  Could you give me some uber simple examples perhaps?  Or see whether/where my examples are wrong? Sometimes that helps ... ?

Reply:
> So for `<component name="myName">` the name attribute value was the reference, and

?? The `name` attribute value is not a reference. Just the name of the component.

> for `<connection component_1="myName" ...>` the value of the component_1 attribute was the reference.

Here, I agree, the `component_1` attribute value is indeed a reference to the component which name is `myName`.

Reply:
I suspect the confusion might be trying to group `component` and `import component` together. A `component` is a thing. Whereas an `import component` is simultaneously both a `component` thing and a reference to a different thing depending on the context.

Reply:
Done in dc25db5